recipes/openssl/1.x.x/conanfile.py

```diff
@@ -535,27 +550,44 @@ def _configure_args(self):
             if self.options.get_safe("no_zlib"):
                 args.append("no-zlib")
             else:
-                zlib_info = self.deps_cpp_info["zlib"]
-                include_path = zlib_info.include_paths[0]
-                if self.settings.os == "Windows":
-                    lib_path = "%s/%s.lib" % (zlib_info.lib_paths[0], zlib_info.libs[0])
+                if Version(conan_version).major < 2:
+                    zlib_info = self.deps_cpp_info["zlib"]
+                    include_path = zlib_info.include_paths[0]
+                    if self.settings.os == "Windows":
+                        lib_path = "%s/%s.lib" % (zlib_info.lib_paths[0], zlib_info.libs[0])
+                    else:
+                        lib_path = zlib_info.lib_paths[0]  # Just path, linux will find the right file
                 else:
-                    lib_path = zlib_info.lib_paths[0]  # Just path, linux will find the right file
+                    zlib_cpp_info = self.dependencies["zlib"].cpp_info
```

Comment:
So the `conan_version` is always something that catches attention, I brought this one up to the team and we discussed if this was the "current best approach" (obviously a moving target - what you did is absolutely correct 👍 we'd just like to set a baseline for others who come after)

This particular line should work in both 1.x and 2.0 - it should be use in the `generate()` save some files and then be read from the `build()` (this is ho the Toolchains` work for comparison)

We'd really love to see that approach used here in this case as well as we things it's the best solution

Reply:
Hi @prince-chrismc,
I implemented what I think you were suggesting here. Let me know if it is okay now.